libc/src/sys/mman/linux/shm_common.h

```diff
@@ -0,0 +1,51 @@
+//===---------- Shared implementations for shm_open/shm_unlink ------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/__support/CPP/array.h"
+#include "src/__support/CPP/optional.h"
+#include "src/__support/CPP/string_view.h"
+#include "src/errno/libc_errno.h"
+#include "src/string/memchr.h"
+#include "src/string/memcpy.h"
+#include <asm/errno.h>
```

Comment:
If you're including `<asm/errno.h>` for the definition of ENAMETOOLONG, I think you should add a define for that to include/llvm-libc-macros/generic-error-number-macros.h then include that here, rather than `<asm/errno.h>`.

Reply:
I probably just remove the include. `libc_errno.h` actually transitively defines `ENAMETOOLONG`